src/test/java/seedu/letsgethired/model/InternTrackerTest.java

```diff
@@ -105,6 +105,25 @@ public void setCurrentApplication_validInternApplication_success() {
         assertTrue(internTracker.getSelectedApplication() == JANE_STREET);
     }
 
+    @Test
+    public void removeApplication_validKeySelectedApplication_success() {
+        InternTracker internTracker = new InternTracker();
+        internTracker.addApplication(JANE_STREET);
+        internTracker.setSelectedApplication(JANE_STREET);
+        internTracker.removeApplication(JANE_STREET);
+        assertFalse(internTracker.hasApplication(JANE_STREET));
+    }
+
+    @Test
+    public void removeApplication_validKeyNotSelectedApplication_success() {
```

Comment:
Wait sorry, but what does this test ah actually? Is that even possible that the User is removing an application but selecting something else?